### ISSUE TITLE
Crash with denied FakePlayer permissions (EnderIO Farmer)

### ISSUE THREAD

**Issue by NitroxydeX** [ForgeEssentials/ForgeEssentialsMain#1578]
Issue: SleepyTrousers/EnderIO#2575

Seems to be related to FE, because after I removed FE all worked fine

**Comment by luacs1998** [comment]
Please provide full FML log.

**Comment by NitroxydeX** [comment]
Here we go: https://dl.dropboxusercontent.com/u/41302965/fml-server-latest.log

**Comment by luacs1998** [comment]
I'll only have time to look into this in a week - on holiday now.

**Comment by NitroxydeX** [comment]
@luacs1998 Maybe @olee could have a look at this if it's related to world-protection. Seems to happen when you right click the farmer.

**Comment by olee** [comment]
@NitroxydeX I cannot see any crash in that log - did you maybe upload the wrong log file?

**Comment by NitroxydeX** [comment]
@olee No, I replaced it for Player with an other logfile. :S My Fail

**Comment by olee** [comment]
Well still - looking at SleepyTrousers/EnderIO#2575 I cannot see how this cannot be their problem.
Also I was unable to reproduce this.
You need to define in more detail what you did.

**Comment by olee** [comment]
REALLY give us proper information when you post such a crash!
This problem only occurs when using the farming station with TinkerConstruct OreBerries and at the same time denying `fe.protection.interact.TConstruct.ore.berries.*`.
The problem seems to be that when the interaction is denied, forge wants to send the player a notification. But because the farmer uses a FakePlayer, it cannot send net-messages to that player, making Minecraft crash.

**Comment by olee** [comment]
Ok after fully checking this issue, I found only two possible solutions:
1. EnderIO needs to change [PickableFarmer.java#Line40](https://github.com/SleepyTrousers/EnderIO/blob/master/src/main/java/crazypants/enderio/machine/farm/farmers/PickableFarmer.java#L40) to not use activateBlockOrUseItem and instead directly use block.onBlockActivated (bad fix)
2. Forge needs a PR to properly check if `thisPlayerMP.playerNetServerHandler` in `activateBlockOrUseItem` is actually null and not try to send an update packet in this case.

TL;DR: This is actually a bug of Forge itself because it cannot handle cancelled events of FakePlayers in activateBlockOrUseItem.

**Comment by NitroxydeX** [comment]
@olee I only can give you the Informations I had. This was: Farmer + Natura Berry Bushes, not Tinkers Construct Bushes. Thats what the User told me. So sorry. There was no crash without FE, only thing I figured out so we thoughts it's a FE-Problem. I set the Flag in the group.

**Comment by olee** [comment]
Once you allowed the correct permissions, it will not crash any more - that's one thing I can guarantee you. You can try harvesting some berries by hand while you have `/p debug` or `/protectdebug` active to see the checked `fe.protection.interact.XXX` permission. This one must be allowed for FakePlayers / all users.

**Comment by NitroxydeX** [comment]
@olee The "NaturaBerryFarmer" ist mod-compat from EnderIO to Natura Berry Bushes. It's not a Item :P

**Comment by olee** [comment]
You know that's already waaaay too late?

**Comment by NitroxydeX** [comment]
@olee was just to let you know.

Set the flag for fakeplayers. Still crashing. grr

**Comment by olee** [comment]
I will test it later - for now just do 
```
/p global allow fe.protection.interact.TConstruct.ore.berries.*
```

**Comment by luacs1998** [comment]
> 2. Forge needs a PR to properly check if thisPlayerMP.playerNetServerHandler in activateBlockOrUseItem is actually null and not try to send an update packet in this case.

I will look into doing this as a part of #1686

**Comment by luacs1998** [comment]
Should we close this issue?
